### PR TITLE
better logs

### PR DESCRIPTION
**Pull request by JarbasAl**
same as #11

### CODE REVIEW

File: ovos_utils/log.py [hunk 1]
```diff
@@ -37,65 +29,97 @@ class LOG:
...
+    def init(cls, config=None):
+        config = config or {}
+        cls.base_path = config.get("path", "stdout")
+        cls.max_bytes = config.get("max_bytes", 50000000)
+        cls.backup_count = config.get("backup_count", 3)
+        cls.level = config.get("level", "DEBUG")
 
     @classmethod
-    def create_logger(cls, name):
+    def create_logger(cls, name, tostdout=True):
```
Comment (ChanceNCounter) [inline ovos_utils/log.py]: `tostdout=False`, surely

File: ovos_utils/log.py [hunk 2]
```diff
@@ -37,65 +29,97 @@ class LOG:
...
+        # [2] - caller
+        stack = inspect.stack()
+
+        # Record:
+        # [0] - frame object
+        # [1] - filename
+        # [2] - line number
+        # [3] - function
+        # ...
+        record = stack[2]
```
Comment (ChanceNCounter) [inline ovos_utils/log.py:98]: Still don't get it. I've never been able to get `inspect.stack()` to do anything remotely useful, nor have I discerned any reliability to the order in which things appear on the stack.

Comment (ChanceNCounter) [reply ovos_utils/log.py:98]: I mean, if it works consistently in Mycroft, and it doesn't crash, I guess it is what it is, but counting on the stack to have exactly what you expect on top seems like *madness*, I don't understand how it's failed to fail for 3+ years

Comment (ChanceNCounter) [reply ovos_utils/log.py:98]: Maybe if this were C++...

**Review by ChanceNCounter — COMMENTED** [review]
Well, I still don't have a way to test this properly, but it looks correct.

File: ovos_utils/log.py (hunk 1 above)
Comment (JarbasAl) [reply ovos_utils/log.py]: disagree, is there any circumstance you don't want it in stdout?

or in other words, do we ever want to log only to file?

i find is very confusing is logs dont show up in stdout during dev...

File: ovos_utils/log.py (hunk 2 above)
Comment (JarbasAl) [reply ovos_utils/log.py:98]: this is immediately the line before this func call, so its consistent. And we extract the module/line where log is happening

i trust @forslund on this one :)

File: ovos_utils/log.py (hunk 1 above)
Comment (ChanceNCounter) [reply ovos_utils/log.py]: During dev, that's true, but isn't the main use case production devices?

**Pull request merged**